### ISSUE TITLE
Pharo-3.0 and Pharo-4.0 fail when #onWarningLog used in configuration

### ISSUE THREAD

**Issue by dalehenrich** [hpi-swa/smalltalkCI#176]
[Here's an example failure](https://travis-ci.org/dalehenrich/metacello-work/jobs/142918392#L248).

This failure is understandable since the version of Metacello present in Pharo3.0 and Pharo4.0 pre-date the addition of #onWarningLog ... We could arrange to have a recent version of Metacello installed, but it is probably not worth the trouble ... 

I'm inclined to use #respondsTo: in the one spot where we send the message in SmalltalkCI and perhaps arrange for Warnings to be handled in a different manner in this case (although this is a bit more difficult to arrange) ...

For now, I'm going to remove the #onWarningLog and see if I can get away with it ...

**Comment by dalehenrich** [comment]
[Also a problem on Squeak-trunk](https://travis-ci.org/dalehenrich/metacello-work/jobs/142918406#L244) ... but this is odd, because I thought Squeak-trunk is fairly recent ... take a look at the log message for this puppy --- looks like an infinite loop ...

**Comment by fniephaus** [comment]
Would the naive approach work? As in: put a block around the load call like this:
```Smalltalk
[ "MC loading" ] on: Exception do: [ :ex | 
      Transcript
        cr;
        show: ex description.
      ex resume ]
```
This would work even if `onWarningLog` is not implemented, wouldn't it?

The reason why `Squeak-trunk` fails is that Metacello preview is not installed. I recently disabled loading it, because it didn't seem to be necessary and saved quite some time. Turns out `Squeak-trunk` does ship with an old version of Metacello. Maybe we should fix that or we need to load Metacello preview again or we could do the above? What do you think?

**Comment by dalehenrich** [comment]
It's never save to have a handler for Exception ... All Notifications are not meant to be handled ... Errors would be swallowed and some exceptions are not resumable ... The following is equivalent to onWarningLog:

```Smalltalk
[ "MC loading" ] on: Warning do: [ :ex | 
      Transcript
        cr;
        show: ex description.
      ex resume ]
```

and is how it's implemented in Metacello ... and would be a portable implementation that would not require having a more recent version of Metacello ... 

If you want to to implement this that would be fine with me :) 

Keep in mind that the Warning handler would need to be conditional on the onWarningLog attribute as not everyone wants to automatically handle Warnings ...

**Comment by dalehenrich** [comment]
I think that the explicit handler is the **right** answer here ... even with the #respondsTo: trick, we have to have a backup implementation using the explicit handler ... might as well just do it unconditionally ...  `onWarningLog` was added as a short cut so that folks didn't have to do their own wrapping ... but it is too convenient to not require the most recent version of Metacello ...

I'll go ahead and prepare a PR for this one ...

**Comment by fniephaus** [comment]
Awesome! Catching only `Warning`s does sound way better than catching all. Thanks!

**Comment by fniephaus** [comment]
Closed by #177
